### ISSUE TITLE
Is it possible to have the size of an ELF file on disk

### ISSUE THREAD

**Issue by LafLaurine** [eliben/pyelftools#451]
Hi, 
I was wondering if it's possible to have the size of an ELF file on the disk, based on the information in its header ?
If I understood well, ` e_shoff + (e_shentsize * e_shnum)` is not always equal to the size on the disk.
Thanks!

**Comment by mzpqnxow** [comment]
Randomly saw this issue at the top of my feed- I'm wondering what you intend to use this for. I ask because for certain use-cases, exposing this as a calculated value based on abstracted header values could be a bit dangerous if not very carefully and clearly documented

I'm thinking specifically of cases involving forensic analysis where malware could take advantage of this behavior to fool analysis logic

I'm not associated with the pyelftools project, so feel free to ignore my question- I'm just curious :)

**Comment by davea42** [comment]
The OP's  reference to `e_shoff + (e_shentsize * e_shnum)` is misguided. That just calculates
the size of the section headers in total.  Referring to those section headers
one gets more relevant data but one has to be aware of the Elf ABI rules.  For example
SHT_NOBITS sh_type has no bits at all though it may have a large sh_size.
All this is well documented on the web.

**Comment by sevaa** [comment]
@LafLaurine No. The ELF format perfectly allows for arbitrary gaps (past the header) and arbitrary trailing space. The loader is not supposed to read it sequentially.

**Comment by sevaa** [comment]
@LafLaurine

Is this still an issue?

**Comment by sevaa** [comment]
@eliben Do you normally insist that whoever opened the issue has to close it?

**Comment by eliben** [comment]
Let's close; the OP can reopen the issue if this is still a problem.

**Comment by sevaa** [comment]
There are other question type issues in the tracker where the OP didn't close and doesn't respond. #352 for example, and more. Mind going over those?

**Comment by eliben** [comment]
> There are other question type issues in the tracker where the OP didn't close and doesn't respond. [#352](https://github.com/eliben/pyelftools/issues/352) for example, and more. Mind going over those?

Sure, I can go over some of these. 

Generally we also have the `pending-user-input` label to mark issues where we're waiting for users to respond, but unfortunately we don't use this very consistently in this project. I will use the label a bit more (and feel free to do so as well), but closing very old issues is also a good option.

**Comment by eliben** [comment]
Here's an example of where old issues still get discovered and can aid some future investigation: https://github.com/eliben/pyelftools/issues/96

I realize that closed issues can be searched also, but in my experience open issues are much easier to find. So the choice of applying a `pending-user-input` label vs. closing an old issue is a heuristic one that should be done on a case-by-case basis.

**Comment by sevaa** [comment]
I can't assign labels.

**Comment by eliben** [comment]
> I can't assign labels.

Sent you an invite to be a collaborator in this repo; use these powers wisely :) I trust your judgement

**Comment by sevaa** [comment]
Thanks.
